nav2_controller/src/controller_server.cpp

```diff
@@ -667,41 +675,55 @@ void ControllerServer::computeAndPublishVelocity()
   RCLCPP_DEBUG(get_logger(), "Publishing velocity at time %.2f", now().seconds());
   publishVelocity(cmd_vel_2d);
 
-  // Find the closest pose to current pose on global path
-  geometry_msgs::msg::PoseStamped robot_pose_in_path_frame;
-  if (!nav2_util::transformPoseInTargetFrame(
-          pose, robot_pose_in_path_frame, *costmap_ros_->getTfBuffer(),
-          current_path_.header.frame_id, costmap_ros_->getTransformTolerance()))
-  {
-    throw nav2_core::ControllerTFError("Failed to transform robot pose to path frame");
-  }
-
-  std::shared_ptr<Action::Feedback> feedback = std::make_shared<Action::Feedback>();
-  feedback->speed = std::hypot(cmd_vel_2d.twist.linear.x, cmd_vel_2d.twist.linear.y);
-
-  nav_msgs::msg::Path & current_path = current_path_;
-  auto find_closest_pose_idx = [&robot_pose_in_path_frame, &current_path]()
+  double current_distance_to_goal_ = 0.0;
+  nav2_msgs::msg::TrackingFeedback current_tracking_feedback_;
+  // Compute and publish tracking error
+  if (!current_path_.poses.empty() && current_path_.poses.size() >= 2) {
+    // Transform robot pose to path frame
+    geometry_msgs::msg::PoseStamped robot_pose_in_path_frame;
+    if (nav2_util::transformPoseInTargetFrame(
+            pose, robot_pose_in_path_frame, *costmap_ros_->getTfBuffer(),
+            current_path_.header.frame_id, costmap_ros_->getTransformTolerance()))
     {
-      size_t closest_pose_idx = 0;
-      double curr_min_dist = std::numeric_limits<double>::max();
-      for (size_t curr_idx = 0; curr_idx < current_path.poses.size(); ++curr_idx) {
-        double curr_dist =
-          nav2_util::geometry_utils::euclidean_distance(robot_pose_in_path_frame,
-          current_path.poses[curr_idx]);
-        if (curr_dist < curr_min_dist) {
-          curr_min_dist = curr_dist;
-          closest_pose_idx = curr_idx;
-        }
+      // Transform goal pose to robot frame for distance calculation
+      geometry_msgs::msg::PoseStamped transformed_end_pose;
+      if (nav2_util::transformPoseInTargetFrame(
+            end_pose_, transformed_end_pose, *costmap_ros_->getTfBuffer(),
+            pose.header.frame_id, costmap_ros_->getTransformTolerance()))
+      {
+        current_distance_to_goal_ = nav2_util::geometry_utils::euclidean_distance(
+          pose, transformed_end_pose);
       }
-      return closest_pose_idx;
-    };
 
-  const std::size_t closest_pose_idx = find_closest_pose_idx();
-  feedback->distance_to_goal = nav2_util::geometry_utils::calculate_path_length(current_path_,
-      closest_pose_idx);
```

Comment:
I'm looking at this block now and there needs to be 2 changes

* Before we threw an exception if the transformation failed that this skips https://github.com/ros-navigation/navigation2/blob/main/nav2_controller/src/controller_server.cpp#L676 Please redo this `if (!transformPoseInTargetFrame(...))` logic to remove the nesting to flatten out this code. 
* shouldn't this already be in the path frame as an element of the path itself? https://github.com/ros-navigation/navigation2/blob/bc84c558415cc01701c616f0c400f279108dd450/nav2_controller/src/controller_server.cpp#L688-L696 I don't think this both is required for `current_distance_to_goal`